### PR TITLE
chore: adding release script

### LINKED ISSUE #167: Automation for the release process
## Issue Description
Automate the process of creating release branches, creating PRs, pushing to docker and finally doing a release on github.

### PR DESCRIPTION
**Pull request by Akarshit**
Resolves #167 
Impact: **minor**  
Type: **chore**

This script aims to automate the release process. In best case scenario, this script should run and do the following
1. Fetches trunk of all repos
4. Updates the CHANGELOG properly
6. Update versions in reaction-development-platform and make a PR for that.

## Breaking changes
None

## Testing
It's really hard to test this one. I have done a couple of releases using this script. To be clear, even if something goes wrong with the script, we only end up with a bad PR for the release.

### CODE REVIEW

**Review by jrw421 — APPROVED** [review]
This looks awesome @Akarshit! As you mentioned, kind of tricky to test, so perhaps we do a tentative approval now and merge after the next release and use that as a test for the script, tweaking as necessary. Lmk what you think!

File: release.py [hunk 1]
```diff
@@ -0,0 +1,474 @@
...
+import os
+from collections import defaultdict
+from git import Repo
+import fileinput
+import time
+from shutil import copyfile
+import sys
+import requests
+
+CORE_MEMBERS = ['akarshitwal@gmail.com', 'jessica.wolvington@gmail.com']
```
Comment (jrw421) [inline release.py]: Think we're missing Mohan's email here

Comment (Akarshit) [reply release.py]: Mohan is using his Mailchimp email for the commits, so he should be fine.

Comment (jrw421) [reply release.py]: Ah, I see. Sounds good!

**Comment by ghost** [comment]
Would much prefer to see something implemented in Bash, if at all, as this project is mostly Node and adding another runtime—which may not be used by many users as it's a pretty personal thing—isn't ideal. Regardless, nice to see some opinion on the release process.

**Comment by Akarshit** [comment]
@balibebas The script is only meant to be used by the core members to do the release. We are putting it in the repo so that it can be version controlled with the platform without creating another repo. The idea behind using python was to get better control and make the code more readable(for me mostly). One things that is part of the plan is to move it to a separate folder so that users who don't need it don't even see it.

**Comment by Akarshit** [comment]
Updated the script after doing a release. Had to make a lot of changes to account for minor inconsistencies.
Should be ready to be merged after doing 1 more release..

**Comment by ghost** [comment]
Is there anyway Docs can be tied into this process so releases also update documentation? As it stands there's no 4.0 release in the docs. Before that they were very consistently released with upgrades AFAICT. https://github.com/reactioncommerce/reaction-docs/issues/965. If not tied in here, some sort of gating process to ensure docs are maintained would be appreciated to help prevent further bitrot.

**Review by MohanNarayana — APPROVED** [review]
👍

**Pull request merged**